### PR TITLE
Rename admission webhooks to avoid name collision

### LINKED ISSUE #723: Rename admission webhooks to avoid name collision
xref: https://github.com/kubedb/project/issues/467

- https://github.com/appscode/stash/blob/master/pkg/controller/replicasets.go#L23
- https://github.com/appscode/stash/blob/master/chart/stash/templates/mutating-webhook.yaml#L105

```
ReplicasetMutator
replicasetmutator
replicasetmutators
```

Apply to both v1alpha1 and v1beta1 apis.

### PR DESCRIPTION
**Pull request by diptadas**
fixes #723

### CODE REVIEW

File: hack/deploy/validating-webhook.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ webhooks:
     service:
       namespace: default
       name: kubernetes
-      path: /apis/admission.stash.appscode.com/v1alpha1/recoveries
+      path: /apis/admission.stash.appscode.com/v1alpha1/recoverievalidators
```
Comment (hossainemruz) [inline hack/deploy/validating-webhook.yaml]: rename `recoverievalidators` -> `recoveryvalidators`

File: pkg/controller/recoveries.go [hunk 2]
```diff
@@ -34,9 +34,9 @@ func (c *StashController) NewRecoveryWebhook() hooks.AdmissionHook {
 		schema.GroupVersionResource{
 			Group:    "admission.stash.appscode.com",
 			Version:  "v1alpha1",
-			Resource: "recoveries",
+			Resource: "recoverievalidators",
```
Comment (hossainemruz) [inline pkg/controller/recoveries.go]: rename `recoverievalidators` -> `recoveryvalidators`

File: chart/stash/templates/validating-webhook.yaml [hunk 3]
```diff
@@ -59,7 +59,7 @@ webhooks:
     service:
       namespace: default
       name: kubernetes
-      path: /apis/admission.stash.appscode.com/v1alpha1/repositories
+      path: /apis/admission.stash.appscode.com/v1alpha1/repositoryvalidators
```
Comment (tamalsaha) [inline chart/stash/templates/validating-webhook.yaml]: deploymentconfigmutators missing from chart

File: pkg/controller/backup_session.go [hunk 4]
```diff
@@ -39,9 +39,9 @@ func (c *StashController) NewBackupSessionWebhook() hooks.AdmissionHook {
 		schema.GroupVersionResource{
 			Group:    "admission.stash.appscode.com",
 			Version:  "v1beta1",
-			Resource: api_v1beta1.ResourcePluralBackupSession,
+			Resource: "backupsessionvalidators",
 		},
-		api_v1beta1.ResourceSingularBackupSession,
+		"backupsessionvalidators",
```
Comment (tamalsaha) [inline pkg/controller/backup_session.go]: backupsessionvalidator[~s~]

**Comment by tamalsaha** [comment]
- scheme for openshift

**Pull request merged**